README_EN.md

```diff
@@ -0,0 +1,240 @@
+
+```Swift
+  let plistPath = Bundle.main.path(forResource: "FRDURLRoutesRegisters", ofType: "plist")
+  let routes = FRDURLRoutes.sharedInstance
+  routes.register(plistFile: plistPath)
+```
+
+Register a block handler.
+
+```Swift
+  let router = FRDURLRoutes.sharedInstance
+  router.register(url: URL(string: "/user/:userId")!) { (params: [String: Any]) in
+    let intent = Intent(url: params[URLRoutes.URLRoutesURL] as! URL)
+    if let topViewController = UIApplication.topViewController() {
+      FRDControllerManager.sharedInstance.startController(source: topViewController, intent: intent)
+    }
+  }
+```
+
+### Get url parameters
+
+FRDIntent/URLRoutes supports simple url parameter pattern match. In the example above, we register url with the pattern `"/story/:storyId"`, if there is an external call like `frdintent://frdintent.com/story/123`, FRDIntent/URLRoutes will save the key `storyId` and the value `123` into the block handler's parameter `params` that is a dictionary. We can get `123` from the dictionary `params` by key `storyId`.
+
+
+## Attention
+
+#### Prefix
+
+Prefix is not necessary in Swift project because of Swift's visiblity declaration. But considering that this library will be used widely in Objective-C project, it's better to add `FRD` prefix to avoid the name conflict in Objective-C.
+
+#### Type of `source` parameter
+
+In the class `FRDControllerManager`, the methods `startController(source: UIVieController, intent: FRDIntent)` and `startControllerForResult(source: UIViewController, intent: FRDIntent, requestCode: Int)` don't constrain the type of `source` parameter. The precise type of `source` is like what does `UIViewController<FRDIntentReceivable>` or `UIViewController<FRDIntentForResultReceivable>` present: it's a specific class conforming a specific protocol. This must be implemented awkwardly by generic type in Swift 3. The generic type is not precise in this situation. And it's more terrrible that the generic method in Swift cann't be used by Objective-C. So we do a tradeoff to declare the source just as UIViewController type. The user of this library must ensure that the source conform the `FRDIntentReceivable` protocol or the `FRDIntentForResultReceivable` protocol by himself.
+
+
+## Intent 和 URLRoutes
+
+FRDIntent/Intent and FRDIntent/URLRoutes can cooperate with each other. FRIntent/Intent handles internal call and FRDIntent/URLRoutes handles external call. In the implementation of FRDIntent/URLRoutes, FRDIntent/URLRoutes only exposes the external call's entry. In the app, FRDIntent/Intent effectively handles the launch process for FRDIntent/URLRoutes.
+
+The separation of FRDIntent/Intent and FRDIntent/URLRoutes is effectively the separation of internal call and external call. There are some advantages of this separation:
+
+- FRDIntent/Intent is better for internal call. It supports to transfer complex data and customize transition animation. Those are hard to be realized in the URL Router's solution.
```

Comment:
- FRDIntent/Intent can be better at handling internal calls. It now supports transmission of complex data and customized transition animation. It will be more difficult to implement those with a simple URL Router solution.